quartz/plugins/emitters/ogImage.tsx

```diff
@@ -114,11 +114,13 @@ export const CustomOgImages: QuartzEmitterPlugin<Partial<SocialImageOptions>> =
       const headerFont = cfg.theme.typography.header
       const bodyFont = cfg.theme.typography.body
       const fonts = await getSatoriFonts(headerFont, bodyFont)
-
-      for (const [_tree, vfile] of content) {
-        if (vfile.data.frontmatter?.socialImage !== undefined) continue
-        yield processOgImage(ctx, vfile.data, fonts, fullOptions)
-      }
+      Promise.all(
+        content.map(([_tree, vfile]) => {
+          if (vfile.data.frontmatter?.socialImage !== undefined) {
+            processOgImage(ctx, vfile.data, fonts, fullOptions)
+          }
+        }),
+      )
```

Comment:
```suggestion
      await Promise.all(
        content.map(([_tree, vfile]) => {
          if (vfile.data.frontmatter?.socialImage !== undefined) {
            yield processOgImage(ctx, vfile.data, fonts, fullOptions)
          }
        }),
      )
```

we still need to wait for the all the promises to finish before returning, also make sure to yield to the generator to report the file as emitted

Reply:
We can't yield at there because it's inside map function and it is not a generator. I made change to entire emit function, not to be async generator but a normal async function which returns promise of array. Does it work? It seems working on my environment.